client/src/modules/libraries/components/License.jsx

```diff
@@ -34,31 +34,32 @@ class License extends PureComponent {
     }
   }
 
-  getLicense = () => {
+  getLicense = async () => {
     const { fetchLicense, libraryName } = this.props;
 
-    this.setState({ pending: true });
-    this.pendingPromise = makeAbortablePromise(fetchLicense(libraryName));
-    this.pendingPromise.promise
-      .then(() => this.setState({ pending: false }))
-      .catch(error => {
-        if (!(error instanceof AbortPromiseException)) {
-          this.setState({ pending: false });
-        }
-      });
+    try {
+      this.setState({ pending: true });
+      this.pendingPromise = makeAbortablePromise(fetchLicense(libraryName));
+      await this.pendingPromise.promise;
+      this.setState({ pending: false });
```

Comment:
you should wait for the result until you set pending to false

Reply:
it is waiting for the result: if you have `await` you don't proceed next lines of code until the awaited code is finished